app/code/Magento/MediaGallery/Test/Unit/Model/Asset/Command/GetByIdTest.php

```diff
@@ -0,0 +1,119 @@
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Framework\DB\Select;
+use Magento\Framework\Exception\IntegrationException;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use Magento\MediaGallery\Model\Asset\Command\GetById;
+use Magento\MediaGalleryApi\Api\Data\AssetInterface;
+use Magento\MediaGalleryApi\Api\Data\AssetInterfaceFactory;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Psr\Log\LoggerInterface;
+use Zend\Db\Adapter\Driver\Pdo\Statement;
+
+/**
+ * Test the GetById command model
+ */
+class GetByIdTest extends TestCase
+{
+    private const MEDIA_ASSET_STUB_ID = 1;
+
+    private const MEDIA_ASSET_DATA = ['id' => 1];
+
+    /**
+     * @var GetById|MockObject
+     */
+    private $getMediaAssetById;
+
+    /**
+     * @var AssetInterfaceFactory|MockObject
+     */
+    private $assetFactory;
+
+    /**
+     * @var AdapterInterface|MockObject
+     */
+    private $adapter;
+
+    /**
+     * @var Select|MockObject
+     */
+    private $selectStub;
+
+    /**
+     * @var Statement|MockObject
+     */
+    private $statementMock;
+
+    /**
+     * @var LoggerInterface|MockObject
+     */
+    private $logger;
+
+    /**
+     * Initialize basic test class mocks
+     */
+    protected function setUp(): void
+    {
+        $resourceConnection = $this->createMock(ResourceConnection::class);
+        $this->assetFactory = $this->createMock(AssetInterfaceFactory::class);
+        $this->logger = $this->createMock(LoggerInterface::class);
+
+        $this->getMediaAssetById = (new ObjectManager($this))->getObject(
+            GetById::class,
+            [
+                'resourceConnection' => $resourceConnection,
+                'assetFactory' => $this->assetFactory,
+                'logger' =>  $this->logger,
+            ]
+        );
+        $this->adapter = $this->createMock(AdapterInterface::class);
+        $resourceConnection->method('getConnection')->willReturn($this->adapter);
+
+        $this->selectStub = $this->createMock(Select::class);
+        $this->selectStub->method('from')->willReturnSelf();
+        $this->selectStub->method('where')->willReturnSelf();
+        $this->adapter->method('select')->willReturn($this->selectStub);
+
+        $this->statementMock = $this->getMockBuilder(\Zend_Db_Statement_Interface::class)->getMock();
+    }
+
+    /**
+     * Test successful get media asset by id command execution.
+     */
+    public function testSuccessfulGetByIdExecution(): void
+    {
+        $this->statementMock->method('fetch')->willReturn(self::MEDIA_ASSET_DATA);
+        $this->adapter->method('query')->willReturn($this->statementMock);
+
+        $mediaAssetStub = $this->getMockBuilder(AssetInterface::class)->getMock();
+        $this->assetFactory->expects($this->once())->method('create')->willReturn($mediaAssetStub);
+
+        $this->getMediaAssetById->execute(self::MEDIA_ASSET_STUB_ID);
```

Comment:
Please, add an assertion here. The `execute` method returns a value, so it worth asserting that we have an expected value.

Reply:
Hi @rogyar I don't think the assertion is necessary here as execute method is not supposed to return anything in this specific test (it is supposed to throw an exception instead)

Reply:
Hi @sivaschenko. In this particular case, we test the positive scenario with no exception. As a result, the `execute` method should return an ID. An assertion of the mentioned ID might be useful here.
Probably, I've missed some points.

Thank you.